provider/import.go

```diff
@@ -52,33 +84,51 @@ func Import(ctx context.Context, p Provider, hcl, tfstate writer.Writer, f *filt
 			}
 
 			if hcl != nil {
+				logger.Log("msg", "calculating HCL")
 				err = r.HCL(hcl)
 				if err != nil {
 					return errors.Wrapf(err, "error while calculating the Config of resource %q", t)
 				}
 			}
 
 			if tfstate != nil {
+				logger.Log("msg", "calculating TFState")
 				err = r.State(tfstate)
 				if err != nil {
 					return errors.Wrapf(err, "error while calculating the Satate of resource %q", t)
 				}
 			}
 		}
+		if resourceLen != 0 {
```

Comment:
I'd even say `> 0`, it's not likely to happen whatsoever but it filters any negative entity.